src/view.c

```diff
@@ -308,13 +310,10 @@ view_store_natural_geometry(struct view *view)
 	 * natural_geometry width/height may still be zero in which case we set
 	 * some fallback values. This is the case with foot and Qt applications.
 	 */
-	if (!view->w || !view->h) {
+	if (wlr_box_empty(&view->current)) {
```

Comment:
This does change the logic, it was ignoring `.x` and `.y` before.
Not sure if that matters though.

Reply:
`wlr_box_empty` ignores x and y too:
https://gitlab.freedesktop.org/wlroots/wlroots/-/blob/master/util/box.c#L37

Reply:
Ohh. I am sorry, should have checked that. Thanks for the pointer.